synapse/storage/databases/main/events.py

```diff
@@ -1766,6 +1766,14 @@ def _store_room_members_txn(
                 self.store.get_invited_rooms_for_local_user.invalidate,
                 (event.state_key,),
             )
+            txn.call_after(
+                self.store.get_local_users_in_room.invalidate,
+                (event.room_id,),
+            )
+            txn.call_after(
+                self.store.get_number_joined_users_in_room.invalidate,
+                (event.room_id,),
+            )
```

Comment:
We need to add a cache invalidation for `get_user_in_room_with_profile` here too